### PR TITLE
Enable no threads in build

### PR DESCRIPTION
**Pull request by jinghe-INTC**
Enable  build configure no-threads from OpenSSL in SGX SSL, with a flag "NO_THREADS"

### CODE REVIEW

File: Linux/build_openssl.sh [hunk 1]
```diff
@@ -74,6 +74,10 @@ if [[ $# -gt 0 ]] && [[ $1 == "debug" || $2 == "debug" || $3 == "debug" || $4 ==
     ADDITIONAL_CONF="-g "
 fi
 
+if [[ $# -gt 0 ]] && [[ $1 == "no-threads" || $2 == "no-threads" || $3 == "no-threads" || $4 == "no-threads" ]] ; then
```
Comment (lzha101) [inline Linux/build_openssl.sh:77]: After adding this option, the max input parameters should be 5. So this check and the above similar checks should be updated.

File: Linux/build_openssl.sh [hunk 2]
```diff
@@ -132,7 +136,7 @@ cp sgx_config.conf $OPENSSL_VERSION/ || exit 1
 cp x86_64-xlate.pl $OPENSSL_VERSION/crypto/perlasm/ || exit 1
 
 cd $SGXSSL_ROOT/../openssl_source/$OPENSSL_VERSION || exit 1
-perl Configure --config=sgx_config.conf sgx-linux-x86_64 --with-rand-seed=none $ADDITIONAL_CONF $SPACE_OPT $MITIGATION_FLAGS no-idea no-mdc2 no-rc5 no-rc4 no-bf no-ec2m no-camellia no-cast no-srp no-hw no-dso no-shared no-ssl3 no-md2 no-md4 no-ui no-stdio no-afalgeng -D_FORTIFY_SOURCE=2 -DGETPID_IS_MEANINGLESS -include$SGXSSL_ROOT/../openssl_source/bypass_to_sgxssl.h --prefix=$OPENSSL_INSTALL_DIR || exit 1
+perl Configure --config=sgx_config.conf sgx-linux-x86_64 --with-rand-seed=none $ADDITIONAL_CONF $SPACE_OPT $MITIGATION_FLAGS no-idea no-mdc2 no-rc5 no-rc4 no-bf no-ec2m no-camellia no-cast no-srp no-hw no-dso no-shared no-ssl3 no-md2 no-md4 no-ui-console no-stdio no-afalgeng -D_FORTIFY_SOURCE=2 -DGETPID_IS_MEANINGLESS -include$SGXSSL_ROOT/../openssl_source/bypass_to_sgxssl.h --prefix=$OPENSSL_INSTALL_DIR || exit 1
```
Comment (lzha101) [inline Linux/build_openssl.sh:139]: This change has been in the master branch, right? Suggest to rebase to latest master branch first.

**Review by lzha101 — CHANGES_REQUESTED** [review]

File: Linux/build_openssl.sh (hunk 2 above)
Comment (jinghe-INTC) [reply Linux/build_openssl.sh]: Done. Thanks

File: Linux/build_openssl.sh (hunk 1 above)
Comment (jinghe-INTC) [reply Linux/build_openssl.sh]: changed to `if [[ "$*" == *"no-threads"* ]] ;`, also for `space-out` and `debug`

**Review by lzha101 — APPROVED** [review]

**Pull request merged**